src/array.vala

```diff
@@ -41,11 +41,8 @@ public class Vast.Array : Object
     /* GObject that owns the memory buffer for storage of scalar elements */
     public Bytes? data {get; construct; }
 
-    /* global shift from the _cached_data; _cached_data + origin is the memory location of 0-th element */
-    public size_t origin { get; construct; default = 0; }
-
-    /* pointer to the begining memory buffer, */
-    internal unowned uint8* _cached_data;
+    /* pointer to the memory location of 0th element */
+    public uint8 * origin {get; construct; default = null;}
```

Comment:
What I don't like here is that we publicly expose a pointer. This won't work if the array is backed by, let's say, a network or a file storage. Couldn't we do the computed property trick around `get_data ()` pointer and have an internal base pointer?